common/determined_common/experimental/checkpoint/_checkpoint.py

```diff
@@ -82,24 +82,31 @@ def __init__(
         self._master = master
 
     def _find_shared_fs_path(self) -> pathlib.Path:
+        """Attempt to find the path of the checkpoint if being configured to shared fs.
+        This function assumes the host path of the shared fs exists.
+        """
         host_path = self.experiment_config["checkpoint_storage"]["host_path"]
         storage_path = self.experiment_config["checkpoint_storage"].get("storage_path")
         potential_paths = [
-            [shared._full_storage_path(host_path, storage_path), self.uuid],
-            [
+            pathlib.Path(shared._full_storage_path(host_path, storage_path), self.uuid),
+            pathlib.Path(
                 shared._full_storage_path(
                     host_path, storage_path, constants.SHARED_FS_CONTAINER_PATH
                 ),
                 self.uuid,
-            ],
+            ),
         ]
 
         for path in potential_paths:
-            maybe_ckpt = pathlib.Path(*path)
-            if maybe_ckpt.exists():
-                return maybe_ckpt
+            if path.exists():
+                return path
 
-        raise FileNotFoundError("Checkpoint {} not found".format(self.uuid))
+        raise FileNotFoundError(
+            "Checkpoint {} not found in {}. We assume the host path of the checkpoint exists"
+            "on the machine where you download checkpoints if using shared file system. "
+            "This error could be caused by not having the same shared file system locally "
```

Comment:
question: do we mean the shared fs needs to be mounted at the same location on the local machine as the agents?